examples/resources/deno_deployment/resource.tf

```diff
@@ -44,4 +44,16 @@ resource "deno_deployment" "example1" {
   env_vars = {
     FOO = "42"
   }
+
+  ###############################################
+  # Custom domain association
+  ###############################################
+  #
+  # Custom domain(s) can be associated with the deployment (optional).
+  # Note the domains must be verified for their ownership and certificates must be ready.
+  # See the doc of deno_domain resource for the full example of the entire process of domain setup.
+
+  # `depends_on` may be useful to ensure the domain is ready.
+  depends_on = [deno_certificate_provisioning.example]
```

Comment:
If we don't do this, does it still work? I would suspect that the lack of verification for the domain would make it not work.

Reply:
The deployment <-> custom domain association API checks if the requested domain is owned by the requester; therefore, at least `depends_on = [deno_domain_verification.example_domain_verification]` is needed to get successful association. However, waiting on `deno_certificate_provisioning` is not a requirement for the association to succeed.
Here I put `depends_on = [deno_certificate_provisioning.example]` because deno_certificate_provisioning in turn depends on deno_domain_verification, so we don't need to explicitly specify the grandchild dependency. But this is a documentation, so I'm wondering what information should be mentioned here